drivers/avr/analog.c

```diff
@@ -128,5 +128,10 @@ int16_t adc_read(uint8_t mux) {
     // Must read LSB first
     low = ADCL;
     // Must read MSB only once!
-    return (ADCH << 8) | low;
+    low |= (ADCH << 8);
+    
+    //turn off the ADC
+    ADCSRA &= ~(1<<ADEN);
```

Comment:
Why does this need to be done?

Reply:
The pin being used as ADC here could be a pin used in the keyboard matrix (hence the saveState/restoreState in process_joystick) so leaving the ADC on could interfere. I'm aware firing up the ADC every time might be a bit costly, how much of a problem do you think this would be?

Reply:
If the pin is already being used in the matrix, it should not be read by the ADC, as the matrix scan is happening continuously and will interfere.

Reply:
As far as I understand, the process call of each feature is sequential, si the matrix scan happens completely, then other features are executed. This is not parallel so there shouldn't be any problem reusing the pins from the matrix to read some analog input. In fact, this is what I did on my keyboard (BminiEx with ATMega32A) and it works fine, given a proper use of diodes to isolate the matrix scanning circuit from the analog reading circuit.